sale_order_lot_selection/models/sale_order_line.py

```diff
@@ -34,6 +37,35 @@ def _compute_product_tracking(self):
         for sol in self:
             sol.product_tracking = sol.product_id.tracking or sol.product_tracking
 
+    @api.depends("product_id", "product_uom_qty", "warehouse_id")
+    def _compute_domain_lot_id(self):
+        dp = self.env["decimal.precision"].precision_get("Product Unit of Measure")
+        for sol in self:
+            domain = []
+            if sol.product_id:
```

Comment:
I don't remember the exact name/values of the field, but for the idea:

```suggestion
            if sol.product_id and sol.product_id.tracking in ("lot", "serial"):
```

Reply:
Condition added, although the `lot_id` field is invisible in this case: https://github.com/OCA/sale-workflow/blob/294b53f3c1a60e50c1050fac68c4de1790ac5a7b/sale_order_lot_selection/views/sale_order_views.xml#L14

Reply:
Yeah, but I would say that the computed is triggered... we can check it just for future references.